pennylane/templates/subroutines/select.py

```diff
@@ -228,3 +234,32 @@ def target_wires(self):
     def wires(self):
         """All wires involved in the operation."""
         return self.hyperparameters["control"] + self.hyperparameters["target_wires"]
+
+
+def _target(target_rep, num_control_wires, state):
+    return qml.decomposition.resources.controlled_resource_rep(
+        base_class=target_rep.op_type,
+        base_params=target_rep.params,
+        num_control_wires=num_control_wires,
+        num_work_wires=0,
+        num_zero_control_values=sum((1 - s for s in state)),
+    )
+
+
+def _select_resources(op_reps, num_control_wires):
+    state_iterator = itertools.product([0, 1], repeat=num_control_wires)
+
+    resources = defaultdict(int)
+    for rep, state in zip(op_reps, state_iterator):
+        resources[_target(rep, num_control_wires, state)] += 1
+    return dict(resources)
+
+
+# pylint: disable=unused-argument
+@qml.register_resources(_select_resources)
+def _select_decomp(*_, wires=None, ops, control, target_wires=None):
+    state_iterator = itertools.product([0, 1], repeat=len(control))
+    return [qml.ctrl(op, control, control_values=state) for state, op in zip(state_iterator, ops)]
```

Comment:
```suggestion
    for state, op in zip(state_iterator, ops):
        qml.ctrl(op, control, control_values=state) 
```
The new decomposition rules shouldn't return anything now.